dbt_sugar/core/task/audit.py

```diff
@@ -0,0 +1,243 @@
+"""Document Task module."""
+from typing import Any, Dict, List
+
+from rich.console import Console
+from rich.table import Table
+
+from dbt_sugar.core.clients.yaml_helpers import open_yaml
+from dbt_sugar.core.flags import FlagParser
+from dbt_sugar.core.logger import GLOBAL_LOGGER as logger
+from dbt_sugar.core.task.base import COLUMN_NOT_DOCUMENTED, BaseTask
+
+console = Console()
+NUMBER_COLUMNS_TO_PRINT_PER_ITERACTION = 5
+
+
+class AuditTask(BaseTask):
+    """Audit Task object.
+
+    Holds methods and attrs necessary to audit a model or a DBT project.
+    """
+
+    def __init__(self, flags: FlagParser) -> None:
+        super().__init__()
+        self.column_update_payload: Dict[str, Dict[str, Any]] = {}
+        self._flags = flags
+        self.model_name = self._flags.model
+
+    def run(self) -> int:
+        """Main script to run the command doc"""
+        if self.model_name:
+            path_file, schema_exists = self.find_model_in_dbt(self.model_name)
+            if not path_file:
+                logger.info("Could not find the Model in the DBT project")
+                return 1
+            if not schema_exists:
+                logger.info("The model is not documented.")
+                return 1
+
+            self.model_content = open_yaml(path_file)
+            self.get_columns_descriptions_statistics_for_model()
+            self.get_columns_tests_statistics_for_model()
+        else:
+            self.get_all_columns_descriptions_statistics_per_model()
+            self.get_all_columns_tests_statistics_per_model()
+        return 0
+
+    def get_columns_tests_statistics_for_model(self) -> None:
+        """
+        Method to get the tests statistics from a model.
+
+        For the process a column is not tested if it does not have any tests.
+        """
+        columns = self.dbt_tests.get(self.model_name)
+
+        if not columns:
+            logger.info(f"Not able to get the test statistics for the model {self.model_name}")
+            return
+
+        # Init variables
+        model_number_columns = 0
+        model_columns_without_tests = 0
+        untested_columns = []
+
+        for column in columns:
+            tests = column["tests"]
+            model_number_columns += 1
+            if len(tests) == 0:
+                model_columns_without_tests += 1
+                untested_columns.append(column["name"])
+
+        percentage_not_tested_columns = round(
+            100 - ((model_columns_without_tests / model_number_columns) * 100), 2
+        )
+
+        # To print nice the data with the total coverage at the last column.
```

Comment:
could you maybe expand this comment? I find that it doesn't really save me from reading and understanding the code and since this is a pretty complex set comprehension it's probably a good idea to make it more human friendly with a comment